src/lib.rs

```diff
@@ -1936,4 +2103,268 @@ mod test {
+        let message = "Those coins belong to Satoshi Nakamoto";
+        let cli_args = vec![
+            "bdk-cli",
+            "--network",
+            "bitcoin",
+            "wallet",
+            "--descriptor",
+            "wpkh(cVpPVruEDdmutPzisEsYvtST1usBR3ntr8pXSyt6D2YYqXRyPcFW)",
+            "verify_proof",
+            "--psbt",
+            psbt.clone(),
+            "--message",
+            message.clone(),
+        ];
+
+        let cli_opts = CliOpts::from_iter(&cli_args);
+
+        let expected_cli_opts = CliOpts {
+            network: Network::Bitcoin,
+            subcommand: CliSubCommand::Wallet {
+                wallet_opts: WalletOpts {
+                    wallet: "main".to_string(),
+                    verbose: false,
+                    descriptor: "wpkh(cVpPVruEDdmutPzisEsYvtST1usBR3ntr8pXSyt6D2YYqXRyPcFW)"
+                        .to_string(),
+                    change_descriptor: None,
+                    #[cfg(feature = "electrum")]
+                    electrum_opts: ElectrumOpts {
+                        timeout: None,
+                        server: "ssl://electrum.blockstream.info:60002".to_string(),
+                        stop_gap: 10,
+                    },
+                    #[cfg(feature = "esplora")]
+                    esplora_opts: EsploraOpts {
+                        esplora: None,
+                        esplora_concurrency: 4,
+                    },
+                    #[cfg(any(feature = "compact_filters", feature = "electrum"))]
+                    proxy_opts: ProxyOpts {
+                        proxy: None,
+                        proxy_auth: None,
+                        retries: 5,
+                    },
+                },
+                subcommand: WalletSubCommand::OnlineWalletSubCommand(VerifyProof {
+                    psbt: Some(psbt.to_string()),
+                    msg: Some(message.to_string()),
+                }),
+            },
+        };
+
+        assert_eq!(expected_cli_opts, cli_opts);
+    }
+
+    /// Encodes a partially signed transaction as base64 and returns the  bytes of the resulting string.
+    #[cfg(all(feature = "reserves", feature = "electrum"))]
+    fn encode_tx(psbt: PartiallySignedTransaction) -> Vec<u8> {
+        let mut encoded = Vec::<u8>::new();
+        psbt.consensus_encode(&mut encoded).unwrap();
+        let tx = base64::encode(&encoded);
+
+        tx.as_bytes().to_vec()
+    }
```

Comment:
just to make the namings a little clear
```suggestion
    fn encode_psbt(psbt: PartiallySignedTransaction) -> Vec<u8> {
        let mut encoded = Vec::<u8>::new();
        psbt.consensus_encode(&mut encoded).unwrap();
        let base64_psbt = base64::encode(&encoded);

        base64_psbt.as_bytes().to_vec()
    }
```